resources/lemmas.md

```diff
@@ -103,8 +103,23 @@ The following lemmas essentially capture the signature extraction mechanisms.
 It reduces the reasoning efforts of the underlying theorem prover, factoring out the essence of the byte-twiddling operations.
 
 ```k
-    rule #padToWidth(32, #asByteStack(V)) => #buf(32, V) // #asByteStackInWidth(V, 32)
-      requires 0 <=Int V andBool V <Int pow256
+    syntax Bool ::= #isRegularWordStack ( WordStack ) [function]
+ // -------------------------------------------------------
+    rule #isRegularWordStack(N : WS => WS)
+    rule #isRegularWordStack(.WordStack) => true
+
+    //Rules for #padToWidth with regular symbolic arguments.
+    //Same as for concrete #padToWidth, when WordStack is of regular form "A:B ... :.WordStack"
+    //Not clear why KEVM rules for #padToWidth were marked [concrete]. If they were general, rules below would not be necessary.
+    rule #padToWidth(N, WS) => WS
+      requires notBool #sizeWordStack(WS) <Int N andBool #isRegularWordStack(WS) ==K true
+
+    rule #padToWidth(N, WS) => #padToWidth(N, 0 : WS)
+      requires         #sizeWordStack(WS) <Int N andBool #isRegularWordStack(WS) ==K true
```

Comment:
i don't think you need all these.

Reply:
i'll remove them and update the pull request accordingly